flink-yarn/src/test/java/org/apache/flink/yarn/TestingYarnAMRMClientAsync.java

```diff
@@ -111,23 +115,35 @@ static Builder builder() {
 		return new Builder();
 	}
 
+	CompletableFuture<Void> getClientInitFuture() {
+		return clientInitFuture;
+	}
+
+	CompletableFuture<Void> getClientStartFuture() {
+		return clientStartFuture;
+	}
+
+	CompletableFuture<Void> getClientStopFuture() {
+		return clientStopFuture;
+	}
+
 	// ------------------------------------------------------------------------
 	//  Override lifecycle methods to avoid actually starting the service
 	// ------------------------------------------------------------------------
 
 	@Override
 	protected void serviceInit(Configuration conf) throws Exception {
-		// noop
+		clientInitFuture.complete(null);
 	}
 
 	@Override
 	protected void serviceStart() throws Exception {
-		// noop
+		clientStartFuture.complete(null);
 	}
 
 	@Override
 	protected void serviceStop() throws Exception {
-		// noop
+		clientStopFuture.complete(null);
 	}
```

Comment:
I think we'd better override `init`, `start` and `stop`, rather than the current `serviceInit`, `serviceStart` and `serviceStop`, to completely prevent behaviors of `AMRMClientAsyncImpl`. The problem is not introduce by this PR though.

Same for `TestingYarnNMClientAsync`.